File: Python/test01.py
# 快速排序算法
def quick_sort(arr):
    # 如果数组长度小于等于1，直接返回数组
    if len(arr) <= 1:
        return arr
    # 选择基准元素
    pivot = arr[len(arr) // 2]
    # 初始化左右子数组
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]
    # 递归排序左右子数组并合并结果
    return quick_sort(left) + middle + quick_sort(right)

# 示例数组
arr = [3,6,8,10,1,2,1]
print("排序前的数组:", arr)

# 调用快速排序函数
sorted_arr = quick_sort(arr)

print("排序后的数组:", sorted_arr)

File: Scala/test01.scala
// 快速排序
def quickSort(arr: Array[Int], left: Int, right: Int): Unit = {
  if (left < right) {
    val pivot = partition(arr, left, right)
    quickSort(arr, left, pivot - 1)
    quickSort(arr, pivot + 1, right)
  }
}

def partition(arr: Array[Int], left: Int, right: Int): Int = {
  val pivot = arr(right)
  var i = left - 1
  for (j <- left until right) {
    if (arr(j) < pivot) {
      i += 1
      swap(arr, i, j)
    }
  }
  swap(arr, i + 1, right)
  i + 1
}

def swap(arr: Array[Int], i: Int, j: Int): Unit = {
  val temp = arr(i)
  arr(i) = arr(j)
  arr(j) = temp
}

val arr = Array(5, 2, 9, 4, 7, 6, 1, 3, 8)
quickSort(arr, 0, arr.length - 1)
println(arr.mkString(", "))